app/controllers/parks_controller.rb

```diff
@@ -61,6 +63,13 @@ def park_params
       :picnic,
       :pets,
       :basketball,
-      :baseball)
+      :baseball
+    )
+  end
+
+  def authorize_user
+    if !user_signed_in? || !current_user.admin?
+      raise ActionController::RoutingError.new("Not Found")
+    end
   end
 end
```

Comment:
Might be better practice to put `def authorize_user` in `application_controller` instead - that way, if you want to authorize someone in another controller, you don't have to redefine this method there